include/SSVOpenHexagon/Core/HexagonGame.hpp

```diff
@@ -262,12 +262,14 @@ class HexagonGame
 
     [[nodiscard]] float getSpeedMultDM() const noexcept
     {
-        return levelStatus.speedMult * (std::pow(difficultyMult, 0.65f));
+        const auto res = levelStatus.speedMult * (std::pow(difficultyMult, 0.65f));
+        return (res < levelStatus.speedMax && levelStatus.speedMax > 0.f) ? res : levelStatus.speedMax;
     }
 
     [[nodiscard]] float getDelayMultDM() const noexcept
     {
-        return levelStatus.delayMult / (std::pow(difficultyMult, 0.10f));
+        const auto res = levelStatus.delayMult / (std::pow(difficultyMult, 0.10f));
+        return std::clamp(res, levelStatus.delayMin, levelStatus.delayMax);
```

Comment:
Should be consistent with the above function, and handle special zero cases for `delayMin` and `delayMax` here?